_delb/nodes.py

```diff
@@ -2780,7 +2780,13 @@ class QueryResults(Sequence[TagNode]):
     """
 
     def __init__(self, results: Iterator[_ElementWrappingNode]):
-        self.__items = cast(Tuple[TagNode], tuple(results))
+        self.__items = cast(Tuple[TagNode, ...], tuple(results))
+
+    def __eq__(self, other):
+        if not isinstance(other, Collection):
+            raise TypeError
+
+        return len(self.__items) == len(other) and all(x in other for x in self.__items)
```

Comment:
tbh I don't understand why this works.. how does python know to correctly determine `x in other` if `other` is of type `QueryResults`, meaning it has to check whether `x in other.__items` even though there is no `QueryResults.__contains__` that tells it that!!?

Reply:
ah [ok](https://docs.python.org/3/library/collections.abc.html#collections-abstract-base-classes).